test/inductor/test_templated_attention.py

```diff
@@ -147,23 +157,116 @@ def score_mod_2(score, b, h, m, n):
 
         self.run_test(composed_score_mod, dtype)
 
-    # TODO We are currently not capturing free variables in the closure correctly
-    @expectedFailure
     @supported_platform
     @common_utils.parametrize("dtype", test_dtypes)
     def test_captured_buffers(self, dtype: torch.dtype):
-        head_offset = torch.rand(8, device="cuda", dtype=dtype)
+        head_offset = torch.rand(H, device="cuda", dtype=dtype)
 
         def score_mod(score, b, h, m, n):
-            return score + head_offset[h]
+            return score + index(head_offset, [h])
```

Comment:
I think we should make a follow up issue to figure out the indexing, maybe tangentially related: https://github.com/pytorch/pytorch/issues/124423